test/built-ins/Promise/prototype/finally/this-value-thenable.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-promise.prototype.finally
+description: >
+  Promise.prototype.finally invoked on thenable returns result of "then" call.
+features: [Promise.prototype.finally]
+---*/
+
+var thenResult = {};
+var Thenable = function() {};
+Thenable.prototype.then = function() { return thenResult; };
+
+assert.sameValue(
+  Promise.prototype.finally.call(new Thenable()),
+  thenResult
```

Comment:
shouldn’t the result of the then call be wrapped in Promise.resolve, not returned directly?

Reply:
Please see step 7 of [Promise.prototype.finally](https://tc39.es/ecma262/#sec-promise.prototype.finally): it forwards to `then` and returns result right away, just like `Promise.prototype.catch` does.

It is `Promise.prototype.then` that wraps in a Promise, but in this test case we have custom `then`:

```js
Thenable.prototype.then = function() { return thenResult; };
```

Reply:
ah right, in order to support subclasses :-) thanks, my mistake